src/app/core/resources/resources.service.js

```diff
@@ -13,261 +13,258 @@ const
-			return Promise.reject({status: 404, message: 'Cannot set tag name to null'});
-		}
+function filterResourcesByAccess(ids, user) {
+	if (!_.isArray(ids)) {
+		return Promise.resolve([]);
+	}
+	else if (null == user.roles || user.roles.admin !== true) {
+		// If user is not admin, perform the filtering
+		return teamsController.filterTeamIds(user).then((teamIds) => {
+			// Get teams user has belongs to
+			teamIds = teamIds.map((teamId) => _.isString(teamId) ? mongoose.Types.ObjectId(teamId): teamId);
 
-		let finalQuery;
-		return constrainTagResults(teamId, user, false).then((query) => {
-			finalQuery = {
+			const query = {
 				$and: [
-					query,
-					{ tags: { $in: [ tagName ] } }
+					{ _id: { $in: ids } },
+					{ $or: [
+						{'owner.type': 'team', 'owner._id': {$in: teamIds}},
+						{'owner.type': 'user', 'owner._id': user._id}
+					]}
 				]
 			};
-			return Resource.updateMany(finalQuery, { $addToSet: { tags: newTagName } }).exec();
-		}).then(() => {
-			return Resource.updateMany(finalQuery, { $pull: { tags: tagName } }).exec();
-		});
-	}
-
-	function deleteTagFromResources(teamId, tagName, user) {
-		if (null == tagName) {
-			return Promise.reject({status: 404, message: 'Invalid tag name'});
-		}
 
-		return constrainTagResults(teamId, user, false).then((query) => {
-			return Resource.updateMany(query, { $pull: { tags: tagName } }).exec();
+			return Resource.find(query).exec();
+		}).then((resources) => {
+			return (null != resources) ? Promise.resolve(resources.map((resource) => resource._id)) : Promise.resolve([]);
 		});
 	}
-
-	function filterResourcesByAccess(ids, user) {
-		if (!_.isArray(ids)) {
-			return Promise.resolve([]);
-		}
-		else if (null == user.roles || user.roles.admin !== true) {
-			// If user is not admin, perform the filtering
-			return teamsController.filterTeamIds(user).then((teamIds) => {
-				// Get teams user has belongs to
-				teamIds = teamIds.map((teamId) => _.isString(teamId) ? mongoose.Types.ObjectId(teamId): teamId);
-
-				const query = {
-					$and: [
-						{ _id: { $in: ids } },
-						{ $or: [
-							{'owner.type': 'team', 'owner._id': {$in: teamIds}},
-							{'owner.type': 'user', 'owner._id': user._id}
-						]}
-					]
-				};
-
-				return Resource.find(query).exec();
-			}).then((resources) => {
-				return (null != resources) ? Promise.resolve(resources.map((resource) => resource._id)) : Promise.resolve([]);
-			});
-		}
-		else {
-			return Promise.resolve(ids);
-		}
+	else {
+		return Promise.resolve(ids);
 	}
+}
 
-	function resourceById(id) {
-		return Resource.findOne({ _id: id }).exec();
-	}
+function resourceById(id) {
+	return Resource.findOne({ _id: id }).exec();
+}
 
-	function find(query, projection, lean) {
-		let promise = Resource.find(query, projection).exec();
-		if (lean) {
-			promise = promise.lean();
-		}
-		return promise;
+function find(query, projection, lean) {
+	let promise = Resource.find(query, projection).exec();
+	if (lean) {
```

Comment:
@jrassa we may want to switch the order of operations here in order to guarantee that the `lean` parameter is used. I _think_ the exec() may execute the query with all of the mongoose middleware trimmings, regardless of whether `.lean()` is called later.